robot/packages/cloud_package_manager.go

```diff
@@ -353,11 +378,34 @@ func sanitizeURLForLogs(u string) string {
 	return parsed.String()
 }
 
-func (m *cloudManager) downloadPackage(ctx context.Context, url string, p config.PackageConfig) error {
-	// TODO(): validate integrity of directory.
-	if dirExists(p.LocalDataDirectory(m.packagesDir)) {
-		m.logger.Debug("Package already downloaded, skipping.")
-		return nil
+// checkNonemptyPaths returns true if all required paths are present and non-empty.
+// This exists because we have no way to check the integrity of modules *after* they've been unpacked.
+// (We do look at checksums of downloaded tarballs, though). Once we have a better integrity check
+// system for unpacked modules, this should be removed.
+func checkNonemptyPaths(dataDir, packageName string, logger logging.Logger, paths []string) bool {
+	missingOrEmpty := 0
+	for _, nePath := range paths {
+		// note: os.Stat treats symlinks as their destination. os.Lstat would stat the link itself.
+		info, err := os.Stat(path.Join(dataDir, nePath))
+		if err != nil {
+			missingOrEmpty++
+			if !os.IsNotExist(err) {
+				logger.Warnw("ignoring non-NotExist error for required path", "path", nePath, "package", packageName, "error", err.Error())
+			}
+		} else if info.Size() == 0 {
```

Comment:
[opt/flyby] Are we worried about people setting their exe path to a symlink to another file in their binary? (probably not)

If we are, we might want to add a check to the filetype.

If not, feel free to resolve without comment.

Reply:
that case is covered in tests -- os.Stat will follow symlinks

(`os.Lstat` will examine links directly)